fix(VirtualizedMessageList): use memoized values as hook dependencies directly

MartinCupela opened the pull request:
### 🎯 Goal

Avoid unnecessary re-renders caused by regenerating a new array of message group style keys (`Object.keys()` was called directly in the hook dependency array).

petyosi reviewed (changes requested):
I believe that we had a bug where `processedMessages` were incorrectly rebuilt with a new object identity at some point, hence the `.length` usage (which, of course, is not correct but we are not aware of any issues because of it). 
Given that, I would say that this change should not touch on that.

MartinCupela commented:
> I believe that we had a bug where `processedMessages` were incorrectly rebuilt with a new object identity at some point, hence the `.length` usage (which, of course, is not correct but we are not aware of any issues because of it). Given that, I would say that this change should not touch on that.

@petyosi I have reverted the changes and added the explanatory comment for future reference.

arnautov-anton commented on src/components/MessageList/VirtualizedMessageList.tsx:
```diff
@@ -206,6 +206,7 @@ const VirtualizedMessageListWithContext = <
         if (style) acc[message.id] = style;
         return acc;
       }, {} as Record<string, GroupStyle>),
```
```suggestion
      }, {}),
```
Not part of the PR so feel free to ignore this comment but noticed this casting which could be refactored to this:
```ts
array.reduce<Record<string, GroupStyle>>(...)
```

MartinCupela replied:
@arnautov-anton added generic type def to all reduce calls

petyosi commented:
:tada: This PR is included in version 10.1.0 :tada:

The release is available on:
- [GitHub release](https://github.com/GetStream/stream-chat-react/releases/tag/v10.1.0)
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/stream-chat-react/v/10.1.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
